### PR TITLE
#1505 Restrict IA upload to Word files only

### PR DESCRIPTION
**Pull request by mbrookeswebdev**
Users should be able to upload only .doc/.docx files for IAs on the Exercise Downloads page:

Screenshot 2021-10-15 at 10 46 15

Other downloads should not be affected:

Screenshot 2021-10-15 at 10 45 24

### CODE REVIEW

File: src/packages/draftComponents/Form/FileUpload.vue [hunk 1]
```diff
@@ -79,6 +80,11 @@ export default {
         };
       },
     },
+    type: {
+      type: String,
+      required: false,
+      default: 'other',
+    },
```
Comment (warrensearle) [inline src/packages/draftComponents/Form/FileUpload.vue:87]: Hey @mbrookeswebdev I agree with @HalcyonJAC comment below. Can we change this prop to be `types`, say, which accepts a CSV string of extensions. Something like the following should do the trick:
```
types: {
  type: String,
  required: false,
  default: '.pdf,.docx,.doc,.odt,.txt,.fodt'
}
```
As a follow on task we should also consider checking mime type.

**Review by warrensearle — CHANGES_REQUESTED** [review]

File: src/packages/draftComponents/Form/FileUpload.vue (hunk 1 above)
Comment (mbrookeswebdev) [reply src/packages/draftComponents/Form/FileUpload.vue]: I experimented with changing the mime type of a file and the browser accepted it, so I think if mime types to be checked, it needs to be done at the back end. Do we need to go that far, given that up to now there was a couple of occasions the staff uploaded the wrong file?

File: src/packages/draftComponents/Form/FileUpload.vue [hunk 2]
```diff
@@ -102,6 +108,12 @@ export default {
         }
       },
     },
+    fileExtensions() {
```
[the remark this answers is not in the document]
Comment (mbrookeswebdev) [reply src/packages/draftComponents/Form/FileUpload.vue:111]: I've made changes as requested.

File: src/packages/draftComponents/Form/FileUpload.vue (hunk 1 above)
Comment (HalcyonJAC) [reply src/packages/draftComponents/Form/FileUpload.vue]: @mbrookeswebdev I think this is OK, mime types might be worthwhile if this turns into a problem or if it needs to be a specific type.

**Review by HalcyonJAC — APPROVED** [review]

**Review by tomlovesgithub — APPROVED** [review]

**Review by warrensearle — APPROVED** [review]
👍

**Pull request merged**